src/compiler/moduleNameResolver.ts

```diff
@@ -1868,14 +1873,19 @@ function loadModuleFromFileNoImplicitExtensions(extensions: Extensions, candidat
     }
 }
 
-function loadJSOrExactTSFileName(extensions: Extensions, candidate: string, onlyRecordFailures: boolean, state: ModuleResolutionState): PathAndExtension | undefined {
+function loadFileName(extensions: Extensions, candidate: string, onlyRecordFailures: boolean, state: ModuleResolutionState): PathAndExtension | undefined {

```

Comment:
Yeah, this name was already bad even before adding a JSON config file lookup to it. I tried to come up with a better one in my refactors but gave up. The distinctive thing about this function is it’s only ever called with paths written in package.json files—never module specifiers written in source files—and so consequently it always allows the candidate to end with a TS extension (but will also try substituting a JS extension for a TS extension). I guess my best suggestion is

```suggestion
/**
 * This function is it’s only ever called with paths written in package.json files - never
 * module specifiers written in source files - and so consequently it always allows the
 * candidate to end with a TS extension (but will also try substituting a JS extension for
 * a TS extension).
 */
function loadFileNameFromPackageJsonField(extensions: Extensions, candidate: string, onlyRecordFailures: boolean, state: ModuleResolutionState): PathAndExtension | undefined {
```

but maybe @sandersn has naming advice as well.

Reply:
Thanks for the great suggestion 👍🏻 , I was thinking about the name of this function and decided to use the general name `loadFileName` :).